### ISSUE TITLE
Conflation fails for clipped Port Au Prince data when stats are turned on.

### ISSUE THREAD

**Issue by mjeffe98** [ngageoint/hootenanny-ui#270]
When you conflate PortAuPrince_OSM_clip2 with PortAuPrince_MGCP_clip2 and set Collect Statistics = false the conflation job completes but if you set it to false, it fails with the error: "2/25/2016, 2:53:29 PM: Requested job failed! For detailed log goto Manage->Log"

The error in Catalina.out is below: 

2016-02-25 14:52:02,160 ERROR DbUtils:1523 - ERROR: relation "current_way_nodes_23" does not exist
  Position: 31
2016-02-25 14:53:27,258 ERROR JobResource:216 - Failed to execute.Error running conflate:
Unknown geometry type: Relation = associatedStreet
make: *** [step1] Error 255

2016-02-25 14:53:27,259 ERROR JobExecutionManager:225 - Job with ID: 603a9077-feac-4865-92a9-d78c14bccfbf failed: {"chainjobstatus":"603a9077-feac-4865-92a9-d78c14bccfbf","childrencount":"3","children":[{"id":"7643062b-4ded-42b2-beeb-6cf58ddb16cb","detail":"Failed to execute.Error running conflate:\nUnknown geometry type: Relation = associatedStreet\nmake: *** [step1] Error 255\n","status":"failed"}]}
[PortAuPrince_MGCP_subset2.zip](https://files.zenhub.io/56cf87cb596ff08d3a7b63a5)
Uploaded using [**ZenHub.io**](http://www.zenhub.io)
[PortAuPrince_OSM_subset2.zip](https://files.zenhub.io/56cf87d0596ff08d3a7b63a6)
Uploaded using [**ZenHub.io**](http://www.zenhub.io)

**Comment by mjeffe98** [comment]
The two datasets are attached to this ticket in the first comment.

![stats failure](https://cloud.githubusercontent.com/assets/7560096/13337538/cec8217c-dbd0-11e5-9a93-5818f9710872.gif)

**Comment by mjeffe98** [comment]
The data were clipped from the original source data so not sure if that has anything do with the issue.  @brianhatchl @sisskind ideas?  Not sure how to proceed with this other than get someone to reproduce it in debug mode.

**Comment by brianhatchl** [comment]
This seems like a genuine conflate error, not related to the stats option
>Error running conflate:
Unknown geometry type: Relation = associatedStreet

@mattjdnv have any input?

**Comment by mjeffe98** [comment]
I suppose what is perplexing is why the error only occurs when setting Conflation Stats = true as opposed to false.  Does setting it to false allow conflation to run in a more permissive mode?

**Comment by brianhatchl** [comment]
@bmarchant I know you've been in the stats code recently, could you have a look at this?

**Comment by bmarchant** [comment]
@brianhatchl I'll take a look at this right now and let you know what I find

**Comment by mjeffe98** [comment]
@bmarchant I just ran 
`hoot conflate PortAuPrince_OSM_subset2.osm PortAuPrince_MGCP_subset2.osm --stats`

and got error below: 

:02:09.085 INFO  ...pp/hoot/core/cmd/ConflateCmd.h( 129) bytesRead: 116765
15:02:09.487 WARN  ...cpp/hoot/core/io/OsmReader.cpp( 537) Missing node (8736) in relation (1).
15:02:09.487 WARN  ...cpp/hoot/core/io/OsmReader.cpp( 537) Missing node (8736) in relation (2).
15:02:09.487 WARN  ...cpp/hoot/core/io/OsmReader.cpp( 537) Missing node (8737) in relation (3).
15:02:09.488 WARN  ...cpp/hoot/core/io/OsmReader.cpp( 537) Missing node (8738) in relation (3).
15:02:09.488 WARN  ...cpp/hoot/core/io/OsmReader.cpp( 537) Missing node (8739) in relation (4).
15:02:09.488 WARN  ...cpp/hoot/core/io/OsmReader.cpp( 537) Missing node (8740) in relation (4).
15:02:09.488 WARN  ...cpp/hoot/core/io/OsmReader.cpp( 537) Missing node (8741) in relation (5).
15:02:09.488 WARN  ...cpp/hoot/core/io/OsmReader.cpp( 537) Missing node (8742) in relation (6).
15:02:09.488 WARN  ...cpp/hoot/core/io/OsmReader.cpp( 537) Missing node (8743) in relation (6).
15:02:09.488 WARN  ...cpp/hoot/core/io/OsmReader.cpp( 537) Missing node (8744) in relation (7).
15:02:09.488 WARN  ...cpp/hoot/core/io/OsmReader.cpp( 541) Found 10 missing nodes, no longer reporting missing nodes.
15:02:09.517 INFO  ...pp/hoot/core/cmd/ConflateCmd.h( 142) inputBytes: 2.98581e+06
15:02:09.517 INFO  .../core/ops/CalculateStatsOp.cpp(  95) Calculating map statistics for input map 1...
Reprojecting 8000 / 8735
Error running conflate:
Unknown geometry type: Relation = restriction

**Comment by bmarchant** [comment]
Here is what I've found so far:
In the function `CalculateStatsOp::apply` on line 326 we apply a `TranslatedTagCountVisitor` that eventually tries to call `ElementConverter::convertToGeometry` on the relations, which calls `ElementConverter::getGeometryType`.  `getGeometryType` is called with the second argument `throwError` set to true so when any relations that aren't explicitly `Relation::isMultiPolygon`, `OsmSchema::isArea`, `OsmSchema::isLinear`, or `OsmSchema::isCollection` an error is thrown.  In looking through the data in PortAuPrince_OSM_subset2.osm there are five relation types that cause errors `duplicate-object`, `restriction`, `associatedStreet`, `boundary`, and `site`.

Long story short, if I change the call to `getGeometryType` in `ElementConverter.cpp` line 140 from:
```
  GeometryTypeId gid = getGeometryType(e, true, statsFlag);
```
to:
```
  GeometryTypeId gid = getGeometryType(e, false, statsFlag);
```
The stats operation on conflation completes successfully.  I've run all of the slow tests on this change and everything checks out.  I'm running glacial now...

**Comment by bmarchant** [comment]
Everything above only happens when collecting stats and has nothing to do with the actual conflation itself.  I'm wondering how long this has been an issue and when it was introduced because I'm assuming that this data has been conflated before without issue.

**Comment by brianhatchl** [comment]
We only recently enabled generation of conflation stats from the web ui, which is probably why we're only noticing this now.

**Comment by bwitham** [comment]
this was handled in ngageoint/hootenanny#418
